lib/article_status_manager.rb

```diff
@@ -12,25 +12,20 @@ def initialize(wiki = nil)
   # Entry points #
   ################
 
-  # Queries deleted state and namespace for all articles from current courses.
-  def self.update_article_status
-    threads = Course.current
-                    .in_groups(Replica::CONCURRENCY_LIMIT, false)
-                    .map.with_index do |course_batch, i|
-      Thread.new(i) do
-        course_batch.each do |course|
-          update_article_status_for_course(course)
-        end
-      end
-    end
-    threads.each(&:join)
-  end
-
   def self.update_article_status_for_course(course)
     Wiki.all.each do |wiki|
       next unless course.pages_edited.exists?(wiki_id: wiki.id)
-      course.pages_edited.where(wiki_id: wiki.id).find_in_batches do |article_batch|
-        new(wiki).update_status(article_batch)
+      # Updating only those articles which are updated more than  1 day ago
+      course.pages_edited
+            .where(wiki_id: wiki.id)
+            .where('articles.updated_at < ? OR articles.created_at = articles.updated_at',
+                   1.day.ago)
+            .in_batches do |article_batch|
+        # Using in_batches so that the update_at of all articles in the batch can be
+        # excuted in a single query, otherwise if we use find_in_batches, query for
+        # each article for updating the same would be required
+        new(wiki).update_status(article_batch.to_a)
```

Comment:
Why is `.to_a` being used here?

Reply:
I thought of giving the same type of input to update_status as before, i.e., `array` (returned by `find_in_batches`), but it seems like `ActiveRecord::Relation` object (returned by `in_batches`) also works the same as array in this case. Removed `to_a`.